tests/queries/0_stateless/02763_row_policy_storage_merge.sql.j2

```diff
@@ -0,0 +1,137 @@
+DROP TABLE IF EXISTS 02763_merge_log_1;
+DROP TABLE IF EXISTS 02763_merge_log_2;
+DROP TABLE IF EXISTS 02763_merge_merge_1;
+DROP TABLE IF EXISTS 02763_merge_merge_2;
+DROP TABLE IF EXISTS 02763_merge_fancycols;
+DROP ROW POLICY IF EXISTS 02763_filter_1 ON 02763_merge_log_1;
+DROP ROW POLICY IF EXISTS 02763_filter_2 ON 02763_merge_merge_1;
+DROP ROW POLICY IF EXISTS 02763_filter_3 ON 02763_merge_log_1;
+DROP ROW POLICY IF EXISTS 02763_filter_4 ON 02763_merge_merge_1;
+DROP ROW POLICY IF EXISTS 02763_filter_5 ON 02763_merge_fancycols;
+DROP ROW POLICY IF EXISTS 02763_filter_6 ON 02763_merge_fancycols;
+
```

Comment:
Could you please write in a description to this PR a very simple example how it worked before your PR and how it's going to work from now on?

Reply:
I have made it by myself:

```
CREATE TABLE tbl1 (x UInt32) ENGINE=MergeTree ORDER BY tuple();
INSERT INTO tbl1 SELECT number AS x FROM numbers(10);
CREATE ROW POLICY r1 ON tbl1 USING x%2==0 TO ALL;
SELECT * FROM tbl1; -- applies the row policy
SELECT * FROM merge('default', 'tbl1'); -- doesn't apply the row policy
```

Reply:
Original behavior is described in the issue
https://github.com/ClickHouse/ClickHouse/issues/50163
which is referred.

Reply:
My example is shorter :)